spec/syntax.md

```diff
@@ -229,272 +314,145 @@ A message can have multiple selectors.
+An _operand_ MAY optionally be followed by an _annotation_.
 
 ```abnf
-variant = when 1*(s key) [s] pattern
-key     = literal / "*"
+operand    = literal / variable
 ```
 
-A _well-formed_ message is considered _valid_ if the following requirements are satisfied:
-
-- The number of keys on each _variant_ MUST be equal to the number of _selectors_.
-- At least one _variant's_ keys MUST all be equal to the catch-all key (`*`).
-- Each _selector_ MUST have an _annotation_,
-  or contain a _variable_ that directly or indirectly references a _declaration_ with an _annotation_.
-
-### Patterns
+### Annotation
 
-A **_pattern_** is a sequence of translatable elements.
-Patterns MUST be delimited with `{` at the start, and `}` at the end.
-A _pattern_'s contents MAY be empty.
-Whitespace within a _pattern_ is meaningful and MUST be preserved.
-This serves 3 purposes:
-
-- The message can be unambiguously embeddable in various container formats
-  regardless of the container's whitespace trimming rules.
-  E.g. in Java `.properties` files,
-  `hello = {Hello}` will unambiguously define the `Hello` message without the space in front of it.
-- The message can be conveniently embeddable in various programming languages
-  without the need to escape characters commonly related to strings, e.g. `"` and `'`.
-  Such need might still occur when a single or double quote is
-  used in the translatable content.
-- The syntax needs to make it as clear as possible which parts of the message body
-  are translatable and which ones are part of the formatting logic definition.
+An **_annotation_** is part of an _expression_ containing either 
+a _function_ together with its associated _options_, or
+a _private-use_ or _reserved_ sequence.
 
 ```abnf
-pattern = "{" *(text / expression) "}"
+annotation = (function *(s option)) / reserved / private-use
 ```
 
-> **Example**
-> 
-> A simple _pattern_ containing _text_:
-> ```
-> {Hello, world!}
-> ```
->
-> An empty _pattern_:
-> ```
-> {}
-> ```
->
-> Some _patterns_ with _expressions_:
-> ```
-> {{$foo}}
-> {Hello {$user}!}
-> {You sent {$count :number maxFractionDigits=0} notifications to {$numFriends :number type=spellout} friends.}
-> ```
->
-> A _pattern_ containing three spaces:
-> ```
-> {   }
-> ```
+### Function
 
-### Expressions
+A **_function_** is a used to evaluate, format, select, or otherwise process an _operand_,
+or, if lacking an _operand_, its _options_.
+A _function_ accepts only an _operand_ as a positional argument.
 
-An _expression_ can appear as a local variable value, as a _selector_, and within a _pattern_.
-The contents of each _expression_ MUST start with an _operand_ or an _annotation_.
-An _expression_ MUST NOT be empty.
+A _function_ consists of a prefix sigil followed by a _name_.
+The following sigils are used for _functions_:
+* `:` for standalone content
+* `+` for starting or opening _expressions_
+* `-` for ending or closing _expressions_
```

Comment:
Further down you define _opening element_ and _closing element_. Should those terms be used here as well?